Applications/Scale/test/testScale.cpp

```diff
@@ -460,4 +470,329 @@ void scalePhysicalOffsetFrames()
+
+        // Transform of custom2 should have been scaled using the scale factors
+        // corresponding to body2.
+        const OpenSim::Function& fn2 =
+            custom2->getSpatialTransform()[3].getFunction();
+        ASSERT_EQUAL(fn2.calcValue(xA), yA * scaleFactors[0],
+            SimTK::SignificantReal, __FILE__, __LINE__,
+            "CustomJoint 'custom2' was incorrectly scaled.");
+        ASSERT_EQUAL(fn2.calcValue(xB), yB * scaleFactors[0],
+            SimTK::SignificantReal, __FILE__, __LINE__,
+            "CustomJoint 'custom2' was incorrectly scaled.");
+    }
+
+    // Test CoordinateCouplerConstraint scaling.
+    {
+        cout << "- CoordinateCouplerConstraint" << endl;
+
+        Model* model = new Model();
+        OpenSim::Body* body1 = new OpenSim::Body("body1", 1, Vec3(0), Inertia(0));
+        model->addBody(body1);
+        OpenSim::Body* body2 = new OpenSim::Body("body2", 1, Vec3(0), Inertia(0));
+        model->addBody(body2);
+
+        // Attach bodies to Ground with slider joints.
+        SliderJoint* slider1 =
+            new SliderJoint("slider1", model->getGround(), *body1);
+        slider1->updCoordinate().setName("indepCoord");
+        model->addJoint(slider1);
+
+        SliderJoint* slider2 =
+            new SliderJoint("slider2", *body2, model->getGround());
+        slider2->updCoordinate().setName("depCoord");
+        model->addJoint(slider2);
+
+        // Constrain Coordinate of slider2.
+        OpenSim::Array<std::string> indepCoordNameArray("indepCoord", 1, 1);
+        const double slope     = 1.2;
+        const double intercept = 3.4;
+
+        CoordinateCouplerConstraint* ccc = new CoordinateCouplerConstraint();
+        ccc->setIndependentCoordinateNames(indepCoordNameArray);
+        ccc->setDependentCoordinateName("depCoord");
+        ccc->setFunction(LinearFunction(slope, intercept));
+        model->addConstraint(ccc);
+
+        // Scale the model with non-uniform scale factors. Should fetch the
+        // scale factors corresponding to body2 and then throw because only
+        // uniform scaling is currently supported.
+        State& s = model->initSystem();
+        ASSERT_THROW(OpenSim::Exception, model->scale(s, scaleSet, false));
+
+        // Test coupling function before scaling.
+        slider1->getCoordinate().setValue(s, 0., true);
+        ASSERT_EQUAL(slider2->getCoordinate().getValue(s), intercept,
+            SimTK::SignificantReal, __FILE__, __LINE__,
+            "SliderJoint has incorrect Coordinate value before scaling.");
+
+        slider1->getCoordinate().setValue(s, 1., true);
+        ASSERT_EQUAL(slider2->getCoordinate().getValue(s), intercept + slope,
+            SimTK::SignificantReal, __FILE__, __LINE__,
+            "SliderJoint has incorrect Coordinate value before scaling.");
+
+        // Scale body2 uniformly.
+        const double uniformFactor = 3.14159;
```

Comment:
I think this is fine here, but why not use a more random number?

Reply:
What random number would you like me to use?

Reply:
I've replaced with 1.234, which I used on L351 and presumably didn't cause confusion. Let me know if you want me to use a seeded random number generator.

Reply:
I prefer typing the numbers in rather than using a random number generator (even if it's seeded) to make debugging easier. Introducing a random number generator also increases the complexity of the test code (potentially introducing issues related only to the random number generator) and reduces readability. I typically use the decimal expansion of pi, the golden ratio, etc. for "random" (arbitrary) numbers, but I recognize that others may not pick up on this and could think that these arbitrary numbers were chosen purposefully. I have therefore replaced the arbitrary numbers in these tests with 1.1, 2.22, 3.333, 0.123, 0.456, 0.789, 1.234, 5.678, 123.456, and 1.23456. These numbers were chosen to convey that they are arbitrary while maintaining the property that they are not multiples of each other (which are often bad numbers to use in test cases). Let me know if these numbers don't align with your definition of "a more random number".

Reply:
I could have been clearer; I did not mean to use a random number generator function. Your recent change is exactly in line with what I was thinking.